packages/eas-cli/src/worker/upload.ts

```diff
@@ -77,7 +78,19 @@ export async function uploadAsync(params: UploadParams): Promise<UploadResult> {
     );
   }
 
-  const contentType = mime.getType(path.basename(params.filePath));
+  let contentType = mime.getType(path.basename(params.filePath));
+
+  if (!contentType) {
+    const fileContent = await readFile(filePath, 'utf-8');
+    try {
+      // check if file is valid JSON without an extension, e.g. for the apple app site association file
+      const parsedData = JSON.parse(fileContent);
+      if (parsedData) {
+        contentType = 'application/json';
+      }
+    } catch {}
+  }
```

Comment:
nit: let's extract this function, so we can prepare for it to be moved into `assets.ts` into an earlier phase.

We could also do a quick check such as `/^\s*[{[]/` to determine whether it's worth calling `JSON.parse`, in case the JSON asset is quite large.

Reply:
As discussed, the pre-check is probably not necessary as `JSON.parse` should fail quite quickly for non-JSON files.

Moved the util to a separate function 👍